pymc/logprob/utils.py

```diff
@@ -201,18 +203,120 @@ def __str__(self):
         return f"Check{{{self.msg}}}"
 
 
-@node_rewriter(tracks=[CheckParameterValue])
+class InnerGraphRewriter:
+    def transform(self, node: Apply) -> Union[Variable, str, None]:
+        """
+        Return: "remove" if node should be removed
+                new_node if node should be replaced by new_node
+                None if node should not be replaced
+        """
+        raise NotImplementedError
+
+    def rewrite(self, node: Union[Scan, OpFromGraph]):
```

Comment:
Just to keep with the API used by other PyTensor rewrites

```suggestion
    def apply(self, fgraph: FunctionGraph, node: Union[Scan, OpFromGraph]):
```